<launch>
  
  <arg name="nao_ip" default="$(optenv NAO_IP 192.168.0.199)"/>
  <arg name="nao_port" default="$(optenv NAO_PORT 9559)"/>

  <!--
    Top camera 2
  -->
  <node pkg="nao_driver" type="nao_camera.py" name="nao_camera2" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <remap from="image_raw" to="image_raw2"/>
  </node>
  
  <!--
    IMAGE CONVERTER 2
  -->
  <node pkg="nao_behavior_tree" type="imageconverter" name="im_conv2" args='2' output="screen">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <!--
    Sonar 2
  -->
  <node pkg="nao_behavior_tree" type="nao_sonar" name="nao_sonar2" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
    <remap from="sonar" to="sonar2"/>
  </node>
  
  <!--
    Leds 2
  -->
  <node pkg="nao_behavior_tree" type="nao_leds" name="nao_leds2" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <!--
    Walker 2
  -->
  <node pkg="nao_driver" type="nao_walker.py" name="nao_walker2" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" output="screen">
  	<remap from="cmd_vel" to="cmd_vel2"/>
  </node>
  
  
  <!--
    Actions 2
  -->
  <node pkg="nao_behavior_tree" type="Greet" name="Greet2" args="2">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="Search" name="Search2" args="2" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="GoClose" name="GoClose2" args="2" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="Stop" name="Stop2" args="2">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="StopWalking" name="StopWalking2" args="2">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="TakeBall" name="TakeBall2" args="2" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>

  <node pkg="nao_behavior_tree" type="Catch" name="Catch2" args="2" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>

</launch>
